pkg/virt-controller/liveupdate/memory.go

```diff
@@ -0,0 +1,145 @@
+	}
+
+	if vm.Spec.Template.Spec.Domain.Memory.Guest != nil && vmi.Status.Memory.GuestAtBoot != nil &&
+		vm.Spec.Template.Spec.Domain.Memory.Guest.Cmp(*vmi.Status.Memory.GuestAtBoot) == -1 {
+		vmConditions := controller.NewVirtualMachineConditionManager()
+		vmConditions.UpdateCondition(vm, &v1.VirtualMachineCondition{
+			Type:               v1.VirtualMachineRestartRequired,
+			LastTransitionTime: metav1.Now(),
+			Status:             k8score.ConditionTrue,
+			Message:            "memory updated in template spec to a value lower than what the VM started with",
+		})
+		return nil
+	}
+
+	// If the following is true, MaxGuest was calculated, not manually specified (or the validation webhook would have rejected the change).
+	// Since we're here, we can also assume MaxGuest was not changed in the VM spec since last boot.
+	// Therefore, bumping Guest to a value higher than MaxGuest is fine, it just requires a reboot.
+	if vm.Spec.Template.Spec.Domain.Memory.Guest != nil && vmi.Spec.Domain.Memory.MaxGuest != nil &&
+		vm.Spec.Template.Spec.Domain.Memory.Guest.Cmp(*vmi.Spec.Domain.Memory.MaxGuest) == 1 {
+		vmConditions := controller.NewVirtualMachineConditionManager()
+		vmConditions.UpdateCondition(vm, &v1.VirtualMachineCondition{
+			Type:               v1.VirtualMachineRestartRequired,
+			LastTransitionTime: metav1.Now(),
+			Status:             k8score.ConditionTrue,
+			Message:            "memory updated in template spec to a value higher than what's available",
+		})
+		return nil
+	}
+
+	memoryDelta := resource.NewQuantity(vm.Spec.Template.Spec.Domain.Memory.Guest.Value()-vmi.Status.Memory.GuestCurrent.Value(), resource.BinarySI)
+
+	newMemoryReq := vmi.Spec.Domain.Resources.Requests.Memory().DeepCopy()
+	newMemoryReq.Add(*memoryDelta)
+
+	// checking if the new memory req are at least equal to the memory being requested in the handleMemoryHotplugRequest
+	// this is necessary as weirdness can arise after hot-unplugs as not all memory is guaranteed to be released when doing hot-unplug.
+	if newMemoryReq.Cmp(*vm.Spec.Template.Spec.Domain.Memory.Guest) == -1 {
+		newMemoryReq = *vm.Spec.Template.Spec.Domain.Memory.Guest
+		// adjusting memoryDelta too for the new limits computation (if required)
+		memoryDelta = resource.NewQuantity(vm.Spec.Template.Spec.Domain.Memory.Guest.Value()-newMemoryReq.Value(), resource.BinarySI)
+	}
+
+	patches := []string{
+		fmt.Sprintf(`{ "op": "test", "path": "/spec/domain/memory/guest", "value": "%s"}`, vmi.Spec.Domain.Memory.Guest.String()),
+		fmt.Sprintf(`{ "op": "replace", "path": "/spec/domain/memory/guest", "value": "%s"}`, vm.Spec.Template.Spec.Domain.Memory.Guest.String()),
+		fmt.Sprintf(`{ "op": "test", "path": "/spec/domain/resources/requests/memory", "value": "%s"}`, vmi.Spec.Domain.Resources.Requests.Memory().String()),
+		fmt.Sprintf(`{ "op": "replace", "path": "/spec/domain/resources/requests/memory", "value": "%s"}`, newMemoryReq.String()),
```

Comment:
Please use the new patchSet type

Reply:
I didn't because it's already done here https://github.com/kubevirt/kubevirt/pull/11655.